met-api/migrations/versions/88aba309bc23_add_engagement_slug.py

```diff
@@ -30,7 +31,35 @@ def upgrade():
     sa.UniqueConstraint('engagement_id'),
     sa.UniqueConstraint('slug')
     )
-    op.create_index('idx_slug', 'engagement_slug', ['slug'], unique=False)
+    op.create_index('idx_slug', 'engagement_slug', ['slug'], unique=False)  
+
+    # Retrieve engagement names that don't have slugs
+    connection = op.get_bind()
+    result = connection.execute("""
+        SELECT id, name
+        FROM engagement
+    """)
+    
+    engagements_without_slugs = result.fetchall()
+
+    # Generate unique slugs for each engagement
+    slugify = SlugGenerationService.create_custom_unique_slugify()
+    for engagement in engagements_without_slugs:
+        engagement_id = engagement[0]
+        engagement_name = engagement[1]
+        slug = slugify(engagement_name)
+
+        # Insert the generated slug into the engagement_slug table
+        connection.execute(
```

Comment:
you can always use op.bulk_insert here.
need not change..just for awareness

Reply:
thanks for noticing, replaced it with bulk insert